src/main/java/com/google/sps/data/Places.java

```diff
@@ -145,17 +149,63 @@
     return stream;
   }
 
+  /**
+   * This function getAllPlaces retrieved from a Places Search API using the API
+   * and the user's location.
+   * @return Set<PlacesSearchResult[]> This returns a set of arrays of Places Search Results.
+   */
+  public static Set<HiddenGem> getHiddenPlaces(Set<PlacesSearchResult> places) {
+    GeoApiContext context = new GeoApiContext.Builder()
+      .apiKey(GetConfigProperties.getApiKey())
+      .build();
+
+    return convertToHiddenGem(context, places);
+  }
+
+  /**
+   * This converts a set of places into a set of hidden gems. 
+   * Iterate over the set of places and find the additional details, then create
+   * the hidden gem object by populating all the relevant data in the constructor.
+   * @param places            The set of places search results to be converted.
+   * @return Set<HiddenGem>   This returns a set of Hidden Gems with all information populated
+   *                          from the place search and place details results. 
+   */
+  public static Set<HiddenGem> convertToHiddenGem(GeoApiContext context, Set<PlacesSearchResult> places) {
+
+    Set<HiddenGem> hiddenGems = new HashSet<>();
+
+    for(PlacesSearchResult place : places){
+      try {
+        PlaceDetails placeDetails = PlacesApi.placeDetails(context, place.placeId)
+        .fields(
+            PlaceDetailsRequest.FieldMask.FORMATTED_ADDRESS,
+            PlaceDetailsRequest.FieldMask.PRICE_LEVEL,
+            PlaceDetailsRequest.FieldMask.WEBSITE)
+        .await();
+
+        hiddenGems.add(new HiddenGem(place.placeId, place.name, place.types,  placeDetails.formattedAddress, place.geometry.location.lat, 
```

Comment:
Nit - Maybe make a line for each parameter? I would make it more readable, but I understand if it is too much

Reply:
done :)